### ISSUE TITLE
Entities set in relationships do not seem to get saved.

### ISSUE THREAD

**Issue by blueneogeo** [OnyxDevTools/onyx-database-parent#72]
```xtend

	@Test
	def void test() {
		val person1 = db.save( new Person => [
			firstName = 'Sally'
			lastName = 'Sad'
		])
		val person2 = db.save( new Person => [
			firstName = 'Jason'
			lastName = 'Bourne'
		])
		db.save( new Address => [
			street = 'Bourbon street'
			houseNr = 1
			occupants = #[ person1, person2 ]
		])
		val results = db.query(Address.Data).list
		println(results.size) // returns 1
		println(results.head.street) // returns 'Bourbon street'
		println(results.head.occupants) // returns empty list, no relationships saved?
	}

```

The expression #[ person1, person2 ] gets translated by Xtend to Java code: 

```java
// google guava Collections
Collections.<Person>unmodifiableList(CollectionLiterals.<Person>newArrayList(person1, person2))
```

The code above uses the following entities:

 ```xtend

@OnyxFields
@OnyxJoins
@Accessors
@Entity
class Address extends ManagedEntity {

	@Identifier(generator=SEQUENCE)
	@Attribute
	Long id

	@Attribute(nullable=false)
	String street

	@Attribute(nullable=false)
	int houseNr

	@Relationship(type=ONE_TO_MANY, inverseClass=Person, inverse='address', cascadePolicy=ALL, fetchPolicy=LAZY)
	List<Person> occupants

	override toString() '''«id»: «street» «houseNr», occupants: «occupants?.map[toString]»'''

}


@OnyxFields
@OnyxJoins
@Accessors
@Entity
class Person extends ManagedEntity {

	@Identifier(generator=SEQUENCE)
	@Attribute
	Long id

	@Attribute(nullable=false)
	Date dateCreated = new Date

	@Attribute
	Date dateUpdated

	@Attribute(nullable=false)
	String firstName

	@Attribute
	int age

	@Attribute(nullable=false)
	String lastName

	@Relationship(type=MANY_TO_ONE, inverseClass=Address, inverse='occupants', cascadePolicy=ALL)
	Address address

	override toString() '''«id»: «firstName» «lastName» age: «age»'''

}

```

**Comment by timbob1000** [comment]
I suspect the println() method uses an iterator().  That may explain why you are not seeing results.  When running this within a pure Java implementation I see that the relationship is persisted as expected.  I also see that the results render 2 relationship objects.

I did however find an issue with the LazyRelationshipCollection iterator().  It was not implemented.  I will supply a merge request to address that.

What happens when you use the equivalent

println(results.head.occupants.get(0))

**Comment by blueneogeo** [comment]
println(results.head.occupants.get(0)) correctly prints "1: Sally Sad age: 0"

**Comment by timbob1000** [comment]
This issue has been addressed, merged into develop and will be released with V1.3.0
